src/vm.rs

```diff
@@ -519,9 +519,17 @@ impl<T: GuestStagePageTable> Vm<T, VmStateFinalized> {
                             }
                         };
 
-                        // We only emulate WFI for now. Everything else gets redirected as an illegal
-                        // instruction exception.
+                        // We only emulate WFI and Crrss for PMU registers for now. Everything else
+                        // gets redirected as an illegal instruction exception.
                         match inst.instruction() {
+                            // Verify that this is a valid PMU CSR supported by the platform, and return
+                            // either the cached value (stopped counter), or the current value.
+                            Instruction::Csrrs(csr_type) if let Some(gpr_index) = GprIndex::from_raw(csr_type.rd()) &&
```

Comment:
I think all these operations should be implemented in a generic fashion so that it can be used very easily across salus.

1. CSR emulation
2. bitmask generation